### PR TITLE
Weapons utilities

### PR DESCRIPTION
**Pull request by wintersleepAI**
Added a few weapon utilities

- Hover for weapon attack/roll shows range and shock
- New config/sys flag for adding shock damage as a "roll" .  Have written a chat damage module for swnr, so want to use for that.
- Added a reload button on the combat tab for weapons that use ammo. This will refill the weapon ammo and give a chat message. Not checking inventory for replacement ammo, but am going to look into.

### CODE REVIEW

File: src/module/settings.ts [hunk 1]
```diff
@@ -20,4 +20,14 @@ export const registerSettings = function (): void {
...
+
+  game.settings.register("swnr", "addShockMessage", {
+    name: "swnr.settings.addShockMessage",
+    hint: "swnr.settings.addShockMessageHint",
+    scope: "world",
+    config: true,
+    type: Boolean,
+    default: true,
+  });
+
```
Comment (Spice-King) [inline src/module/settings.ts:32]: Opposite to the last PR, this should not be an option.
```suggestion
```

File: src/module/actors/character-sheet.ts [hunk 2]
```diff
@@ -237,7 +262,31 @@ export class CharacterActorSheet extends ActorSheet<
...
+            flavor: shock_content,
+            type: CONST.CHAT_MESSAGE_TYPES.ROLL,
+            roll: JSON.stringify(shockRoll.toJSON()),
+          });
+        }
+      }
+
     };
+
+
```
Comment (Spice-King) [inline src/module/actors/character-sheet.ts:289]: Roll this all into the attack roll message. Shock is part of the attack and is not a separate attack, thus should not be sent as an extra message.

File: src/lang/en.yml [hunk 3]
```diff
@@ -366,9 +367,12 @@ swnr:
     base-ac: Base AC
     vs: vs
     rollables: Rollables
+    hit-dice: Hit Dice
   settings:
     useHomebrewLuckSave: Use Homebrew Luck saves?
     useHomebrewLuckSaveHint: Uses 16 - Level for the save, no modifiers, just pure luck
+    addShockMessage: Add Shock Damage Message?
+    addShockMessageHint: When attacking this adds another seperate message log for shock damage. Useful for modules.
```
Comment (Spice-King) [inline src/lang/en.yml:375]: See the comment on the setting code.
```suggestion
```

**Review by Spice-King — CHANGES_REQUESTED** [review]
As a note, I dislike the use of `console.log` being left in places. I keep meaning to use the DevMode module, clean up and consolidate debug logging.

File: src/module/settings.ts (hunk 1 above)
Comment (wintersleepAI) [reply src/module/settings.ts:32]: Sounds like with the discord conversation you were good with this left in as long as the shock was not a separate chat entry. This config will show a roll in the same message for shock if set.

**Comment by wintersleepAI** [comment]
Made the requested changes. With the config flag set shock damage looks like this now:

image

Will need to do some code clean up on the coming macro PR as the damage rolls were refactored some.

**Pull request closed without merging**